### DISCUSSION TITLE
Add AppThemeColor

### DISCUSSION THREAD

**Discussion by PureWeen** [CommunityToolkit/Maui#1261] [New Feature Discussions]
### Description of Change

This adds an AppThemeColor that can be used to drastically simplify creating a color palette and reduce the verbosity needed when creating an AppThemeBinding. **NOTE** This does make a minor deviation from the original proposal as XamlC was thinking the expected type was `AppThemeColorExtension` instead of `AppThemeColor` when declaring the AppThemeColor.

```xml
<!-- APP THEME COLORS -->
<AppThemeColor Light="#B0150303" Dark="#FCFCFC" x:Key="TextPrimaryColor" />

<Style x:Key="Headline" TargetType="Label">
    <Setter Property="FontFamily" Value="Segoe UI" />
    <Setter Property="FontSize" Value="10" />
    <!--<Setter Property="TextColor" Value="{AppThemeBinding Light={StaticResource LightTextPrimaryColor}, Dark={StaticResource DarkTextPrimaryColor}}" />-->
    <Setter Property="TextColor" Value="{AppTheme TextPrimaryColor}" />
</Style>
```

Also added in this PR is `AppThemeResource` which is meant to be a generic version of the `AppThemeColor` that could be use more broadly for things like an Image Source which may need to change based on Light/Dark mode:

```xml
<ResourceDictionary>
  <AppThemeResource Light="MyLightLogo.png" Dark="MyDarkLogo.png" x:Key="MyLogo" />
</ResourceDictionary>

<Image Source="{AppTheme MyLogo}" />
```

### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/2597

**Comment by VladislavAntonyuk** [comment]
I like it, but before moving forward, In order to avoid reoccurring issues, what was the reason to remove it from Xamarin.Forms?

**Reply by PureWeen** [reply]
@jfversluis I'm having a little trouble following the journey of `AppThemeColor` in `Xamarin.Forms`. Did we used to have this in `XF` and then we removed it? It seems like it was around in 4.6 and then removed in 4.7?

**Reply by jfversluis** [reply]
My initial implementation had it. There was the `AppThemeColor` and `OnAppTheme<T>`. Later this was renamed to AppThemeBinding and these helper objects were removed.

Here is [`AppThemeColor`](https://github.com/xamarin/Xamarin.Forms/blob/0ebc41203516cb2dbd01d582038b81deb451a2b1/Xamarin.Forms.Core/AppThemeColor.cs) and here is [`OnAppTheme`](https://github.com/xamarin/Xamarin.Forms/blob/0ebc41203516cb2dbd01d582038b81deb451a2b1/Xamarin.Forms.Core/OnAppTheme.cs) from the history. These are not available in the current version of Xamarin.Forms anymore.

**Reply by jfversluis** [reply]
Created a draft PR with a PoC based on Dan's work here: https://github.com/CommunityToolkit/Maui/pull/1264

Seems like there is nothing blocking us from adding this to the Toolkit in its current form. Just need to see if everyone is on board, add some samples and put it in the right place.

**Comment by TheCodeTraveler** [comment]
Completed https://github.com/CommunityToolkit/Maui/pull/1264
